Moved broadcast logic to BroadcastManager.kt

egorbunovs opened the pull request:
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

egorbunovs committed:
Moved broadcast logic to BroadcastManager.kt

egorbunovs committed:
Fixed race condition bugs with camera exchanging;
Refactored the BroadcastManager.kt - made it more readable and manageable;

egorbunovs committed:
Small update

egorbunovs committed:
Another small fix

egorbunovs committed:
Another small fix

egorbunovs committed:
Yet another small fix

egorbunovs committed:
Fixed broadcast start / end state handling;

slee-aws commented:
The app still freezes for me with the latest update. I can partially reproduce it on a Galaxy S10 with the following steps:
- Reset the app to first-time state (uninstall/reinstall)
- Complete the onboarding
- From the home view in portrait orientation, start a broadcast
- During the broadcast, rotate to landscape orientation and back to portrait

At this point, the app UI becomes unresponsive and the camera shows a black screen. This doesn't seem to happen 100% of the time however, so I'm trying to come up with better steps to reproduce.

egorbunovs committed:
Updated Broadcast SDK version and gradle libs;
Fixed a bug with broadcast start / end state;

egorbunovs commented:
I have a Galaxy S10 and haven't noticed any freezes with the latest code in this branch. Could you please re-verify if that's the same for you @slee-aws ? :)

egorbunovs committed:
Fixed session release / create;
Fixed - selected camera is not re-attached on session restart;
Fixed - stream data must be consumable so that it ignores already consumed events;

egorbunovs committed:
Fixed - screen share flow is not updated;

slee-aws commented:
The changes are working for me now. LGTM - will merge and close this PR. Thank you for the contribution!
